cmd/cstorvolume-claim/cvc/cstorvolumeclaim.go

```diff
@@ -0,0 +1,399 @@
+			err,
+			"failed to convert to int {%s}",
+			count,
+		)
+	}
+
+	spcName := scObj.Parameters["storagePoolClaim"]
+	return rCount, spcName, nil
+}
+
+// listCStorPools get the list of available pool using the storagePoolClaim
+// as labelSelector.
+func listCStorPools(
+	spcName string,
+	replicaCount int,
+) (*apis.CStorPoolList, error) {
+
+	if spcName == "" {
+		return nil, errors.New("failed to list cstorpool: spc missing")
+	}
+
+	labelSelector := "openebs.io/storage-pool-claim=" + spcName
+
+	cstorPoolList, err := csp.KubeClient().List(metav1.ListOptions{
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"failed to list cstorpool for spc {%s}",
+			spcName,
+		)
+	}
+	if len(cstorPoolList.Items) < replicaCount {
+		return nil, errors.New("not enough pools available to create replicas")
+	}
+	return cstorPoolList, nil
+}
+
+// createTargetService creates cstor volume target service
+func createTargetService(storageClassName string,
+	claim *apis.CStorVolumeClaim,
+) (*corev1.Service, error) {
+
+	labels := map[string]string{
+		"openebs.io/target-service":      "cstor-target-svc",
+		"openebs.io/storage-engine-type": "cstor",
+		"openebs.io/persistent-volume":   claim.Name,
+	}
+	selectors := map[string]string{
+		"app":                          "cstor-volume-manager",
+		"openebs.io/target":            "cstor-target",
+		"openebs.io/persistent-volume": claim.Name,
+	}
+	annotations := map[string]string{
+		"openebs.io/storage-class-ref": "name: " + storageClassName,
+	}
+
+	OwnerReference := []metav1.OwnerReference{
+		*metav1.NewControllerRef(claim,
+			apis.SchemeGroupVersion.WithKind("CStorVolumeClaim")),
+	}
+
+	svcObj, err := svc.NewKubeClient(svc.WithNamespace("openebs")).
+		Get(claim.Name, metav1.GetOptions{})
+
+	if err != nil && !k8serror.IsNotFound(err) {
+		return nil, errors.Wrapf(
+			err,
+			"failed to get cstorvolume service {%v}",
+			svcObj.Name,
+		)
+	}
+	if k8serror.IsNotFound(err) {
+		svcObj, err = svc.NewBuilder().
+			WithName(claim.Name).
+			WithLabelsNew(labels).
+			WithAnnotations(annotations).
+			WithOwnerRefernceNew(OwnerReference).
+			WithSelectorsNew(selectors).
+			WithPorts(ports).
+			Build()
+		if err != nil {
+			return nil, errors.Wrapf(
+				err,
+				"failed to build target service {%v}",
+				svcObj,
+			)
+		}
+
+		svcObj, err = svc.NewKubeClient(svc.WithNamespace("openebs")).Create(svcObj)
+	}
+	return svcObj, err
+}
+
+// cstorvolumereplicacr creates CStorVolume resource for a cstor volume
+func createCStorVolumecr(service *corev1.Service,
+	claim *apis.CStorVolumeClaim,
+	scName string,
+) (*apis.CStorVolume, error) {
+
+	labels := map[string]string{
+		"openebs.io/target-service":    "cstor-target-svc",
+		"openebs.io/persistent-volume": claim.Name,
+	}
+	annotations := map[string]string{
+		//"openebs.io/storage-class-ref": "name: " + storageClassName,
+		"openebs.io/fs-type": "ext4",
+		"openebs.io/lun":     "0",
+	}
```

Comment:
Above info is not needed. As the filesystem type is passed to CSI node while mounting the volume.
LUN is constant so there is no need for this annotation also.